libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipRouter.kt

```diff
@@ -594,7 +640,24 @@ open class GossipRouter(
     private fun enqueueIhave(peer: PeerHandler, messageIds: List<MessageId>, topic: Topic) =
         pendingRpcParts.getQueue(peer).addIHaves(messageIds, topic)
 
+    private fun sendIdontwant(peer: PeerHandler, messageId: MessageId) {
+        val iDontWant = Rpc.RPC.newBuilder().setControl(
+            Rpc.ControlMessage.newBuilder().addIdontwant(
+                Rpc.ControlIDontWant.newBuilder()
+                    .addMessageIDs(messageId.toProtobuf())
+            )
+        ).build()
+        send(peer, iDontWant)
+    }
+
     data class AcceptRequestsWhitelistEntry(val whitelistedTill: Long, val messagesAccepted: Int = 0) {
         fun incrementMessageCount() = AcceptRequestsWhitelistEntry(whitelistedTill, messagesAccepted + 1)
     }
+
+    data class IDontWantCacheEntry(
+        val heartbeatMessagesCount: AtomicInteger = AtomicInteger(0),
```

Comment:
nit: Maybe just `var heartbeatMessagesCount: Int`? All the code is thread safe here